File: 732-my-calendar-iii/732-my-calendar-iii.java
class MyCalendarThree {
    
    private TreeMap<Integer, Integer> tmap;
    public MyCalendarThree() {
        tmap = new TreeMap<>();
    }
    
    public int book(int start, int end) {
        
        tmap.put(start, tmap.getOrDefault(start, 0) + 1);
        tmap.put(end, tmap.getOrDefault(end, 0) - 1);
        
        int activeBookings = 0;
        int maxBookings = 0;
        
        for(int events : tmap.values()){
            
            activeBookings += events;
            maxBookings = Math.max(activeBookings, maxBookings);
        }
        
        return maxBookings;
    }
}

/**
 * Your MyCalendarThree object will be instantiated and called as such:
 * MyCalendarThree obj = new MyCalendarThree();
 * int param_1 = obj.book(start,end);
 */

File: 732-my-calendar-iii/732-my-calendar-iii.cpp
class MyCalendarThree {
private:
    map<int, int> m;
public:
    MyCalendarThree() {
        
    }
    
    int book(int start, int end) {
        
        m[start]++;
        m[end]--;
        
        int maxBookings=0;
        int activeBookings=0;
        for(auto it = m.begin(); it != m.end(); it++){
            
            activeBookings += it->second;
            maxBookings = max(maxBookings, activeBookings);
        }
        
        return maxBookings;
    }
};

/**
 * Your MyCalendarThree object will be instantiated and called as such:
 * MyCalendarThree* obj = new MyCalendarThree();
 * int param_1 = obj->book(start,end);
 */